### PR TITLE
Add buildah 0.2 with optional ssl certdir workspace

### PR DESCRIPTION
**Pull request by chmouel**
# Changes

Add buildah 0.2 with optional ssl certdir workspace

We are bumping the buildah task to 0.2 since this needs at least pipeline 0.17.0.

This will add a optional workspace for the cert dir to pass to buildah --cert-dir flag.

This add as well a complete sample on how to use it with the openshift internal registry.

Reworked a bit the test to use pipelineSpec and make it to run against a registry with ssl certificate.

/cc @sbwsg 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [✔️] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [✔️] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [✔️] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [✔️] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [✔️] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [✔️] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [✔️] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [✔️] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [✔️] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by sbwsg** [comment]
Very nice.

/approve

File: task/buildah/0.2/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+[Buildah](https://github.com/projectatomic/buildah) build tool. It uses
+Buildah's support for building from
+[`Dockerfile`](https://docs.docker.com/engine/reference/builder/)s, using its
+`buildah bud` command. This command executes the directives in the `Dockerfile`
+to assemble a container image, then pushes that image to a container registry.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/buildah/0.1/buildah.yaml
```
Comment (vinamra28) [inline task/buildah/0.2/README.md:13]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/buildah/0.2/buildah.yaml
```

File: task/buildah/0.2/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+* **source**: A [Workspace](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md) containing the source to build.
+* **sslcertdir**: An [*optional* Workspace](https://github.com/tektoncd/pipeline/blob/v0.17.0/docs/workspaces.md#optional-workspaces) containing your custom SSL certificates to connect to the registry. Buildah will look for files ending with *.crt, *.cert, *.key into this workspace. See [this sample](./samples/buildah-custom-ssl.yaml) for a complete example on how to use it with OpenShift internal registry.
+
+## Usage
+
+This TaskRun runs the Task to fetch a Git repo, and build and push a container
+image using Buildah.
+
+```
```
Comment (vinamra28) [inline task/buildah/0.2/README.md:44]: so that hub can render this code block properly :sweat_smile: 
```suggestion
```yaml
```

**Review by vinamra28 — COMMENTED** [review]
a few nitpicks :sweat_smile:

**Review by vinamra28 — APPROVED** [review]
/lgtm
looks good...thanks :+1:

**Review by Divyansh42 — APPROVED** [review]
/lgtm

**Pull request merged**